Cabal/Distribution/Simple/Configure.hs

```diff
@@ -606,11 +606,31 @@ configure (pkg_descr0, pbi) cfg = do
             installedPackageSet
             comp
 
+    -- Decide if we're going to compile with split sections.
+    split_sections :: Bool <-
+       if not (fromFlag $ configSplitSections cfg)
+            then return False
+            else case compilerFlavor comp of
+                        GHC | compilerVersion comp >= mkVersion [8,0]
```

Comment:
I think this should be 8.2?

Reply:
8.0 also supports `-split-sections`. We merely started building binary distributions with ``split-sections` in 8.2